javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/extension/ignore/IgnoredTypesConfigurer.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.extension.ignore;
+
+import io.opentelemetry.instrumentation.api.config.Config;
+import io.opentelemetry.javaagent.extension.Ordered;
+
+/**
+ * An {@link IgnoredTypesConfigurer} can be used to override built-in instrumentation restrictions:
+ * ignore some classes and exclude them from being instrumented, or explicitly allow them to be
+ * instrumented if the agent ignored them by default.
+ *
+ * <p>This is a service provider interface that requires implementations to be registered in a
+ * provider-configuration file stored in the {@code META-INF/services} resource directory.
+ */
+public interface IgnoredTypesConfigurer extends Ordered {
```

Comment:
Is there a reason for a separate interface instead of knob on InstrumentationModule?

Reply:
I thought that it'd be useful to be able ignore some classes (e.g. task classes to prevent context leaking) even if you turn the instrumentation off.

Reply:
Hmm - well we could choose to respet them even if it's disabled? Not sure if it's a great idea - IIRC, dd-trace-java has the knob on InstrumentationModule so it could keep us more consistent with them, though maybe that's a difficult goal at this point

Reply:
As far as I understand, they have something similar just for concurrent/task classes: https://github.com/DataDog/dd-trace-java/blob/master/dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/ExcludeFilterProvider.java
And they're doing a separate pass over `Instrumenter`s and check with `instanceof` if this interface is implemented.

> it could keep us more consistent with them, though maybe that's a difficult goal at this point

With all our changes to muzzle, the split to `InstrumentationModule`/`TypeInstrumentation`, and DD's custom things (like profiling) it's probably impossible 😅 

> Hmm - well we could choose to respet them even if it's disabled?

That's what seemed a bit confusing to me - you have the `enabled` flag but it only works for some `InstrumentationModule` methods.